pfio/v2/config.py

```diff
@@ -0,0 +1,50 @@
+import configparser
+import os
+from typing import Dict, Optional
+
+
+def _default_config_file():
+    path = os.getenv('PFIO_CONFIG_PATH')
+    if path:
+        return path
+
+    basedir = os.getenv('XDG_CONFIG_HOME')
+    if not basedir:
+        basedir = os.path.join(os.getenv('HOME'), ".config")
+
+    return os.path.join(basedir, "pfio.ini")
+
+
+def _reload_config():
+    global _config
+    config = configparser.ConfigParser()
+    configfile = _default_config_file()
+    config.read(configfile)
+    _config = config
+
+
+def add_custom_scheme(
+    name: str,
+    scheme: str,
```

Comment:
I don't want to have `scheme` independent input but just have built in a flat dictionary. Thus this should not be just for scheme, but for general configurations and shall be named as `add_config(..)` or `add_config_section(..)` and such. The config file is not just for custom scheme, but for more general configuration (which is not yet to be).

Reply:
I thought this config mechanism is just for custom schemes, because a config key is a scheme name itself.
My assumption is that more fundamental refactoring of low-level config mechanism would be needed to support general config entries. So I think it's better to provide a higher-level `add_custom_scheme` function here.
(I feel no problem about merging `scheme` and `data` arguments though)

How do you think?
That being said, I'm fine with either ways.

Reply:
Yes, I should have expressed my intention more clearly. I understand your reasoning and it's quite correct. Making the section names as custom scheme names was too narrow idea. Maybe we can agree later on how much fundamental refactoring needed for general config entries. Once we got it, this `add_custom_scheme()` can be changed to a shallow wrapper of it.

So my conclusion here would be that, my request is just to add docs here. No need to change function names.

Reply:
No, you don't have to do that, but +1 to check the `scheme`.

Reply:
Thank you. Added docstrings: 61dd14b94eeeaab45ee8a2beee8d2e57f6852a9a